view_conversations/conv-viewer-webapi/src/routes.ts

```diff
@@ -74,27 +74,89 @@ export function setup(
     }
   });
 
-  app.get('/api/toxicity/:upper_score/:lower_score', async (req, res) => {
+  app.get('/api/toxicity/:upper_score/:lower_score/:order/:searchBy/:searchFor/:isAlive', async (req, res) => {
     try {
+      console.log("RECEVED REQUEST")
       const upper_score: number = runtime_types.assertNumber(req.params.upper_score);
       const lower_score: number = runtime_types.assertNumber(req.params.lower_score);
-      const index = conf.spannerTableName + toxicityIndex;
-
+      const isAlive: boolean = runtime_types.assertBoolean(req.params.isAlive);
+      const order: string = runtime_types.assertOrder(req.params.order);
+      let index = conf.spannerTableName + toxicityIndex;
+      if (order === 'ASC') {
+        index = index + '_ASC';
+      }
+      let searchQuery = '';
+      let isAliveStatement = '';
+      switch (req.params.searchBy) {
+        case 'page-id': {
+          const page_id: runtime_types.PageId =
+              runtime_types.PageId.assert(req.params.searchFor);
+          searchQuery = ` and page_id = ${page_id} `;
+          break;
+        }
+        case 'page-title': {
+          const page_title: runtime_types.PageTitleSearch =
+            runtime_types.PageTitleSearch.assert(req.params.searchFor);
+          searchQuery = ` and page_title = "${page_title}" `;
+          break;
+        }
+        case 'user-name': {
+          const user_text: runtime_types.UserTextSearch =
+            runtime_types.UserTextSearch.assert(req.params.searchFor);
+          searchQuery = ` and user_text= "${user_text}" `;
+          break;
+        }
+        case 'user-id': {
+          const user_id: runtime_types.UserId =
+            runtime_types.UserId.assert(req.params.searchFor);
+          searchQuery = ` and user_id = "${user_id}" `;
+          break;
+        }
+        case 'rev-id': {
+          const rev_id: runtime_types.RevisionId =
+            runtime_types.RevisionId.assert(req.params.searchFor);
+          searchQuery = ` and rev_id= ${rev_id} `;
+          break;
+        }
+        case 'comment-id': {
+          const comment_id: runtime_types.CommentId=
+            runtime_types.CommentId.assert(req.params.searchFor);
+          searchQuery = ` and comment_id= ${comment_id} `;
+          break;
+        }
+        case 'conversation-id': {
+          const conversation_id: runtime_types.ConversationId=
+            runtime_types.ConversationId.assert(req.params.ConversationId);
+          searchQuery = ` and conversation_id = ${conversation_id}`;
+          break;
+        }
+      }
+      if (isAlive) {
+        isAliveStatement = ' and isAlive = true';
```

Comment:
I suggest you have a variable for a list of "where_query_conditions", and in each one you don't have the extra " and" string, and then you simply join them. This will be more scalable, and need less variable names. Also it will be more robust to forgetting space characters.